main.tf

```diff
@@ -35,14 +35,14 @@ resource "kubernetes_deployment" "service_pod" {
 
     selector {
       match_labels = {
-        app = "${local.identifier}-service-pod"
+    name      = ${var.service_account_name} == "" ? "${local.identifier}-service-pod" : "${var.service_account_name}-service-pod"
       }
     }
 
     template {
       metadata {
         labels = {
-          app = "${local.identifier}-service-pod"
+    name      = ${var.service_account_name} == "" ? "${local.identifier}-service-pod" : "${var.service_account_name}-service-pod"
```

Comment:
```suggestion
    name      = var.service_account_name == "" ? "${local.identifier}-service-pod" : "${var.service_account_name}-service-pod"
```

Reply:
@markreesmoj just spotted that these suggested changes were using the old interpolation syntax which is no longer supported outside of string expressions (see broken unit test actions). This should work though